### PR TITLE
Change pip to poetry, Makefile for interacting with poetry easier, use dotenv instead of putting it in `bot/config.py` and changing typos in `bot/__main__.py`

### PR DESCRIPTION
**Pull request by timelessnesses**
**What kind of change does this PR introduce?**  
Features

**Describe the changes proposed in the pull request**  
Change from pip to poetry for isolated dependencies from global, makefile for easier to run bots (no need of `poetry shell` and `python bot`, pre installed basic code formatting tools like black and isort and finally using dotenv so you won't commit `bot/config.py` accidentally

**What is the current behavior?**  
Virtualenv is sucks and usually .gitignore won't work if the file is already committed (in my knowledge)

**What is the new behavior**  
Likely better developer experience.

**Does this PR introduce a breaking change?**  
Likely forcing people to use poetry and GNU make (though they still have option to install with `requirements.txt` and start it as normal but needed to edit .env file for it or set enviroment variable since its required by default

**Other information:**  
Also likely none

### CODE REVIEW

**Comment by timelessnesses** [comment]
should i also add automation for code formatting when the commits were uploaded to github?

**Comment by timelessnesses** [comment]
and for anyone concerns about config.py here's is the incident commit.  
https://github.com/v1rbox/discox/pull/48/commits/1f0b6254566a5a6a6359fd522e35c5c60bc78233  
imo its better to contain it inside .env since .gitignore on config.py won't work anyway because of it already tracking it

**Comment by imindMan** [comment]
Uh that just fits your taste. Me likely no using virtualenv for coding because I don't really need that (mess up with your current pip?)

**Comment by timelessnesses** [comment]
> Uh that just fits your taste. Me likely no using virtualenv for coding because I don't really need that (mess up with your current pip?)

**yes**

File: bot/commands/tic_tac_toe.py [hunk 1]
```diff
@@ -1,60 +1,60 @@
-import random
```
Comment (logicguy1) [inline bot/commands/tic_tac_toe.py:1]: This file has been moved to /commands/game/

File: poetry.lock [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+version = "2.0.1"
+description = "A lil' TOML parser"
+category = "dev"
+optional = false
+python-versions = ">=3.7"
+
+[[package]]
+name = "wavelink"
+version = "1.3.4"
+description = "A robust and powerful Lavalink wrapper for discord.py and derivatives."
```
Comment (logicguy1) [inline poetry.lock:220]: What are we using wavelink for?

**Comment by logicguy1** [comment]
@timelessnesses In general looks fine, you mostly updated the formatting of the code witch is good, only thing i belive is missing would be a small setup guide in the README file to get people up and running with the new dependency engine

File: poetry.lock (hunk 2 above)
Comment (timelessnesses) [reply poetry.lock:220]: whoops
poetry didn't remove that
forgot to `poetry lock` 😅

File: bot/commands/tic_tac_toe.py (hunk 1 above)
Comment (timelessnesses) [reply bot/commands/tic_tac_toe.py:1]: that one isn't mine though

Comment (timelessnesses) [reply bot/commands/tic_tac_toe.py:1]: i will move it i guess

**Comment by timelessnesses** [comment]
should be all good now

**Comment by timelessnesses** [comment]
oh uh how can i edit wiki page?

**Comment by logicguy1** [comment]
I had a few comments

**Comment by timelessnesses** [comment]
is this mergable yet?

**Pull request merged**